tests/scalapack/scalapack_06b.cc

```diff
@@ -0,0 +1,156 @@
+  std::vector<Vector<NumberType>> p_eigenvectors_ (max_n_eigenvalues,Vector<NumberType>(size));
+  FullMatrix<NumberType> p_eigenvectors (size,size);
+
+  ScaLAPACKMatrix<NumberType> scalapack_syevx (size, grid, block_size);
+  scalapack_syevx.set_property(LAPACKSupport::Property::symmetric);
+
+  create_spd (full_A);
+  scalapack_syevx = full_A;
+
+  //Lapack as reference
+  {
+    std::vector<NumberType> lapack_A(size*size);
+    for (unsigned int i = 0; i < size; ++i)
+      for (unsigned int j = 0; j < size; ++j)
+        lapack_A[i*size+j] = full_A(i,j);
+
+    int info; //Variable containing information about the successfull exit of the lapack routine
+    char jobz = 'V';  //'V': all eigenpairs of A are computed
+    char uplo = 'U';  //storage format of the matrix A; not so important as matrix is symmetric
+    int LDA = size;   //leading dimension of the matrix A
+    int lwork;      //length of vector/array work
+    std::vector<NumberType> work (1);
+
+    //by setting lwork to -1 a workspace query for work is done
+    //as matrix is symmetric: LDA == size of matrix
+    lwork = -1;
+    syev(&jobz, &uplo, &LDA, & *lapack_A.begin(), &LDA, & *eigenvalues_Lapack.begin(), & *work.begin(), &lwork, &info);
+    lwork=work[0];
+    work.resize (lwork);
+    syev(&jobz, &uplo, &LDA, & *lapack_A.begin(), &LDA, & *eigenvalues_Lapack.begin(), & *work.begin(), &lwork, &info);
+
+    AssertThrow (info==0, LAPACKSupport::ExcErrorCode("syev", info));
+    for (int i=0; i<max_n_eigenvalues; ++i)
+      for (int j=0; j<size; ++j)
+        s_eigenvectors_[i][j] = lapack_A[(size-1-i)*size+j];
+
+  }
+
+  // the actual test:
+
+  pcout << "comparing " << max_n_eigenvalues << " eigenvalues and eigenvectors computed using LAPACK and ScaLAPACK pdsyevx:" << std::endl;
+  const std::vector<NumberType> eigenvalues_psyevx = scalapack_syevx.eigenpairs_symmetric_by_index(std::make_pair(size-max_n_eigenvalues,size-1),true);
+  scalapack_syevx.copy_to(p_eigenvectors);
+  for (unsigned int i=eigenvalues_psyevx.size()-1; i>0; --i)
+    {
+      if ( !(std::abs(eigenvalues_psyevx[i]-eigenvalues_Lapack[size-eigenvalues_psyevx.size()+i]) /
+             std::abs(eigenvalues_Lapack[size-eigenvalues_psyevx.size()+i]) < tol))
+        {
+          std::cout << "process #" << this_mpi_process << ": eigenvalues do not fit: "
+                    << eigenvalues_psyevx[i] << " <--> " << eigenvalues_Lapack[size-eigenvalues_psyevx.size()+i] << std::endl;
+        }
+
+      AssertThrow ( std::abs(eigenvalues_psyevx[i]-eigenvalues_Lapack[size-eigenvalues_psyevx.size()+i]) /
+                    std::abs(eigenvalues_Lapack[size-eigenvalues_psyevx.size()+i]) < tol,
+                    ExcInternalError());
+    }
+  pcout << "   with respect to the given tolerance the eigenvalues coincide" << std::endl;
+
+  // FIXME: run-time error on macOS if code between "pcout << comparing" and this line is executed.
```

Comment:
Does that mean that we can't get run this test on mac?

Reply:
correct, i will investigate further later

Reply:
could be some bug in netlib-scalapack as it works in double precision and also works on all testers so far.

Reply:
Do we still want to merge this? It won't break the tester?

Reply:
yes we do, as I said the current state it works for all testers. This PR **only splits** the test into several parts.